src/maxwell-web-view.js

```diff
@@ -0,0 +1,242 @@
+     * child_draw() is called and we actually have something to show
+     */
+    window.webkit.messageHandlers.maxwell_children_allocate.postMessage(children);
+
+    /* Extra paranoid, update positions if anything changes in the DOM tree!
+     * ideally it would be nice to directly observe BoundingClientRect changes.
+     */
+    update_positions_throttled();
+};
+
+/* Main DOM observer */
+let observer = new MutationObserver(document_mutation_handler);
+observer.observe(document, {
+    childList: true,
+    subtree: true,
+    attributes: true
+});
+
+/* Semi Public API */
+
+/* Main entry point */
+window.maxwell = {}
+
+/* List of children */
+window.maxwell.children = {}
+
+/* child_resize()
+ */
+window.maxwell.child_resize = function (id, width, height) {
+    let child = maxwell.children[id];
+
+    if (!child)
+        return;
+
+    /* Resize canvas keeping the old contents */
+    if (child.width !== width || child.height !== height) {
+        let ctx = child.getContext('2d');
+        var old_image = ctx.getImageData(0, 0, child.width, child.height);
+
+        child.width = width;
+        child.height = height;
+
+        ctx.putImageData(old_image, 0, 0);
+
+        old_image = null;
+    }
+}
+
+/* child_draw()
+ *
+ * Draw child canvas with maxwell:///canvasid image
+ *
+ * Unfortunatelly WebGL does not support texture_from_pixmap but we might be able
+ * to use GL to implement this function if we can get the context from WebKit
+ * itself
+ */
+window.maxwell.child_draw = function (id, image_id, x, y, width, height) {
+    let child = maxwell.children[id];
+
+    if (!child)
+        return;
+
+    let ctx = child.getContext('2d');
+
+    /* Get image data */
+    let xhr = new XMLHttpRequest();
+    let uri = 'maxwell:///' + id;
+    if (image_id)
+        uri += '?' + image_id;
+    xhr.open('GET', uri, false);
+    xhr.responseType = 'arraybuffer';
+
+    try {
+        xhr.send();
+    } catch (error) {
+        return;
+    }
+
+    /* Update child canvas */
+    if (xhr.response) {
+        let data = new Uint8ClampedArray(xhr.response);
+        let image = new ImageData(data, width, height);
+
+        ctx.putImageData(image, x, y);
+
+        /* Help GC */
+        data = null;
+        image = null;
+    }
+
+    xhr = null;
+}
+
+/* child_set_visible()
+ *
+ * Show/hide widget element
+ */
+window.maxwell.child_set_visible = function (id, visible) {
+    let child = maxwell.children[id];
+
+    if (child)
+        child.style.display = (visible) ? child.maxwell_display_value : 'none';
+}
+
+/* child_init()
+ *
+ */
+window.maxwell.child_init = function (id, width, height, visible) {
+    let child = maxwell.children[id];
+
+    if (!child)
+        return;
+
+    child.width = width;
+    child.height = height;
+    window.maxwell.child_draw (id, null, 0, 0, width, height);
+
+    if (visible)
+        window.maxwell.child_set_visible (id, visible);
```

Comment:
Use a consistent JS coding style for whether or not a space exists before braces when calling a method